syntax = "proto3";

package channel.v1;

import "google/protobuf/empty.proto";

option go_package = "github.com/synnaxlabs/synnax/pkg/distribution/transport/grpc/channel/v1";

service ChannelCreateService {
    rpc Exec(CreateMessage) returns (CreateMessage) {}
}

service ChannelDeleteService {
    rpc Exec(DeleteRequest) returns (google.protobuf.Empty) {}
}

service ChannelRenameService {
    rpc Exec(RenameRequest) returns (google.protobuf.Empty) {}
}

message CreateMessage {
    repeated Channel channels = 1;
}

message DeleteRequest {
    repeated uint32 keys = 3;
}

message RenameRequest {
    repeated uint32 keys = 1;
    repeated string names = 2;
}

message Channel {
    string name = 1;
    int32 node_id = 2;
    uint32 storage_key = 3;
    double rate = 4;
    string data_type = 5;
    int32 storage_index = 6;
    bool is_index = 7;
}
